### DISCUSSION TITLE
GraphQL-Java-Tools and Spring for GraphQL

### DISCUSSION THREAD

**Discussion by tedepstein** [graphql-java-kickstart/graphql-java-tools#728] [Q&A]
Looking at the [Spring for GraphQL](https://spring.io/projects/spring-graphql) docs for the first time, I don't see any mention of GraphQL-Java-Tools, and I don't see Resolvers and Schema type bindings in the GraphQL-Java-Tools style. 

But Spring for GraphQL seems to have come up with its own mechanisms for schema-bound classes, and it provides some abstractions over DataFetchers, possibly similar to Resolvers, but I have not read deeply enough to know.

Has anyone thought about whether Spring for GraphQL, intentionally or not, replaces or overlaps with GraphQL-Java-Tools? If GraphQL-Java-Tools offers something important that is not covered by Spring for GraphQL, is there any thought as to how one might integrate GraphQL-Java-Tools into a Spring for GraphQL service?

**Comment by bsara** [comment]
@oliemansm, I have to admit that I was very sad to see that the spring boot project has been archived. I've found it to be so much easier to use than the solution provided by Spring and wished it would keep going (and I do hope that you'll reconsider its archival). That said, it would be nice if we could continue to use this library with Spring's official GraphQL libs. If the plan is to keep this project alive, could we get some example of how we might be able to integrate this lib with the official Spring libs?

**Reply by oliemansm** [reply]
I can understand that, and you might not be the only one. That being said the activity and number of releases particularly of the graphql-spring-boot project had drastically been reduced past two years already. Lack of time and contributors, but also that it doesn't seem to make sense to try to keep a project alive while Spring has more resources to maintain the official library. This project was necessary at first since an official library didn't exist yet, but now we'd rather focus and contribute in other areas instead of trying to compete with an official library.

@oryan-block is the primary maintainer of this `graphql-java-tools` project (which is also used by non-Spring projects) and best to align with him on both contributing and integrating with offical Spring libs. I am wondering though what you think the combination of `graphql-java-tools` with Spring for GraphQL would bring? How would it be different or better compared to just using Spring for GraphQL?

**Reply by bsara** [reply]
> This project was necessary at first since an official library didn't exist yet, but now we'd rather focus and contribute in other areas instead of trying to compete with an official library.

This makes a lot of sense. I wish that they would've adopted the approach taken by you guys though. It's so much more usable...and already has many of the features that they still don't have yet. Did you guys ever have the opportunity to offer graphql-spring-boot features as contributions to Spring for GraphQL? I've never contributed to an official Spring project and can't seem to find a way to get in touch with the maintainers either other than logging issues with the project, which doesn't seem the appropriate avenue for such discussion.

> I am wondering though what you think the combination of graphql-java-tools with Spring for GraphQL would bring? How would it be different or better compared to just using Spring for GraphQL?

This is a good point. It likely wouldn't bring anything at this point, unless it was used under the hood of Spring for GraphQL. I think I was in shock of the closing of the other project when I asked. :)

**Reply by oliemansm** [reply]
I haven't contributed myself, so not sure how open they are to contributions. However they do have a code of conduct which states they're open to it. Haven't read it all but maybe they have certain rules, like first create a feature request issue to get approval to make it part of the lib before you spend a lot of time implementing a PR that won't get merged in the end. See https://github.com/spring-projects/spring-graphql/blob/main/CODE_OF_CONDUCT.adoc

I'd also check the pull request activity. Are there a lot of open PRs and how often do they merge PRs from external contributors? That might be a good indication if they're open to it. 

I'd probably start with a feature request and indicating you're happy to contribute by creating a PR if they're open to it. If the above options don't already give you a clear path forward.

**Reply by oryan-block** [reply]
Unfortunately I can barely keep up with bugs and regular maintenance of this project. I don't really have time to look into what Spring is doing. As far as I know they just decided that this project doesn't jive with their plans so they're just ignoring it.
